src/aklite_client_ext.cc

```diff
@@ -276,3 +277,51 @@ bool AkliteClientExt::RebootIfRequired() {
 
   return false;
 }
+
+InstallResult AkliteClientExt::Rollback(const LocalUpdateSource* local_update_source, int version,
+                                        const std::string& target_name) {
+  auto pending_target = Target::toTufTarget(client_->getPendingTarget());
+  auto current = GetCurrent();
+
+  auto storage = INvStorage::newStorage(client_->config.storage, false);
+  auto update_mode =
+      pending_target.IsUnknown() ? InstalledVersionUpdateMode::kBadTarget : InstalledVersionUpdateMode::kNone;
+
+  // Set current target as failing
+  storage->saveInstalledVersion("", Target::fromTufTarget(current), update_mode);
+
+  // Get rollback target
+  TufTarget rollback_target;
+  if (version == -1 && target_name.empty()) {
+    rollback_target = Target::toTufTarget(client_->getRollbackTarget(true));
```

Comment:
I would change signature of `AkliteClient::GetRollbackTarget()` so it can accept the boolean parameter, or introduce another protected method so we use `GetRollbackTarget()` instead of `client_->getRollbackTarget`.

Reply:
Added. As a general rule, I was trying not to add complexity to the external API.